File: baekjoon/cpp/15652.cpp
#include <iostream>
using namespace std;

int N, M;
int arr[9]{};

void dfs(int level) {
    
    if(level == M) {
        for(int i = 1; i <= M; i++) 
            cout << arr[i] << " ";
        cout << "\n";
        return;
    }
    
    for(int i = 1; i <= N; i++) {
        if(arr[level] > i) continue;
        
        arr[level + 1] = i;
        dfs(level + 1);
    }
}

int main()
{
    cin >> N >> M;
    dfs(0);
}

File: baekjoon/15652.py
n, m = map(int, input().split())
arr = [0] * m

def dfs(cur, level):
    if level == m:
        print(*arr)
        return 
    
    for i in range(cur, n+1):
        arr[level] = i
        dfs(i, level + 1)


dfs(1, 0)